plugins/kernel_v2/config/disk.rb

```diff
@@ -107,21 +113,20 @@ def finalize!
         @size = nil if @size == UNSET_VALUE
         @file = nil if @file == UNSET_VALUE
 
-        @disk_ext = "vdi" if @disk_ext == UNSET_VALUE
-
         if @primary == UNSET_VALUE
           @primary = false
         end
 
-        if @name == UNSET_VALUE
+        if @name.is_a?(String) && @name.match(FILE_CHAR_REGEX)
+            @logger.warn("Vagrant will remove detected invalid characters in '#{@name}' and convert the disk name into something usable for a file")
+            @name.gsub!(FILE_CHAR_REGEX, "")
```

Comment:
When invalid characters are encountered, I think that we would be better to replace them rather than remove them (like with `"_"`). This will prevent things like empty names being generated.